tools/json2cbor/json2cbor.c

```diff
@@ -365,7 +365,7 @@ CborError decode_json(cJSON *json, CborEncoder *encoder)
 
         for (item = json->child ; item; item = item->next) {
             if (usingMetaData && strlen(item->string) > strlen(meta_data_marker)
-                    && strcmp(item->string + strlen(item->string) - 5, meta_data_marker) == 0)
+                    && strcmp(item->string + strlen(item->string) - strlen(meta_data_marker), meta_data_marker) == 0)
```

Comment:
Should we prefer sizeof -1 here?  it IS a constant string...  Could be done above too.

Either way, calculating this 2x in this ifstmt is kinda crappy.

Reply:
strlen(constant) is usually replaced by the compiler with a constant. You of all people should know that :-)